billing/actions/accounts.py

```diff
@@ -220,3 +228,123 @@ def assign_funds_to_invoice(invoice_id: str) -> bool:
             Charge.objects.create(account_id=account_id, amount=-overpayment, product_code=CREDIT_REMAINING)
 
     return invoice.status == Invoice.PAID
+
+
+def mark_accounts_as_delinquent(
+    account_ids: List[int],
+    unpaid_invoices_threshold: Optional[int],
+    days_since_last_unpaid_threshold: Optional[int],
+    currency_amount_threshold_map: Optional[dict],
+) -> Dict[int, str]:
+    """
+    Mark accounts as delinquent when some criteria are accomplished
+    :param account_ids: List of account ids to be evaluated
+    :param unpaid_invoices_threshold: Number of unpaid invoices to consider an user as
+                                      a delinquent
+    :param days_since_last_unpaid_threshold: Days to take into account since the last
+                                             unpaid invoice to consider an user as a
+                                             delinquent
+    :param currency_amount_threshold_map: Balance threshold to consider an user as a
+                                          delinquent.
+                                          Ex: {'CHF': 200, 'EUR': 100, 'NOK': 150}
+    :return: New delinquent accounts and reason Ex: {111: 'Reason why', ...}
+    """
+    legal_accounts = Account.objects.filter(id__in=account_ids, delinquent=False)
```

Comment:
I'd say "compliant" accounts (not "legal")

Reply:
:+1: